x-pack/plugins/apm/server/routes/span_links/get_span_links_details.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { chunk, isEmpty } from 'lodash';
+import {
+  SERVICE_NAME,
+  SPAN_ID,
+  SPAN_NAME,
+  TRACE_ID,
+  TRANSACTION_ID,
+  TRANSACTION_NAME,
+  TRANSACTION_DURATION,
+  SPAN_DURATION,
+  PROCESSOR_EVENT,
+  SPAN_SUBTYPE,
+  SPAN_TYPE,
+  AGENT_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { Environment } from '../../../common/environment_rt';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { SpanLinkDetails } from '../../../common/span_links';
+import { SpanLink } from '../../../typings/es_schemas/raw/fields/span_links';
+import { SpanRaw } from '../../../typings/es_schemas/raw/span_raw';
+import { TransactionRaw } from '../../../typings/es_schemas/raw/transaction_raw';
+import { Setup } from '../../lib/helpers/setup_request';
+import { getBufferedTimerange } from './utils';
+
+async function fetchSpanLinksDetails({
+  setup,
+  kuery,
+  spanLinks,
+  start,
+  end,
+}: {
+  setup: Setup;
+  kuery: string;
+  spanLinks: SpanLink[];
+  start: number;
+  end: number;
+}) {
+  const { apmEventClient } = setup;
+
+  const { startWithBuffer, endWithBuffer } = getBufferedTimerange({
+    start,
+    end,
+  });
+
+  const response = await apmEventClient.search('get_span_links_details', {
+    apm: {
+      events: [ProcessorEvent.span, ProcessorEvent.transaction],
+    },
+    _source: [
+      TRACE_ID,
+      SPAN_ID,
+      TRANSACTION_ID,
+      SERVICE_NAME,
+      SPAN_NAME,
+      TRANSACTION_NAME,
+      TRANSACTION_DURATION,
+      SPAN_DURATION,
+      PROCESSOR_EVENT,
+      SPAN_SUBTYPE,
+      SPAN_TYPE,
+      AGENT_NAME,
+    ],
+    body: {
+      size: 1000,
+      query: {
+        bool: {
+          filter: [
+            ...rangeQuery(startWithBuffer, endWithBuffer),
+            ...kqlQuery(kuery),
+            {
+              bool: {
+                should: spanLinks.map((item) => ({
+                  bool: {
+                    filter: [
+                      { term: { [TRACE_ID]: item.trace.id } },
+                      {
+                        bool: {
+                          should: [
+                            { term: { [SPAN_ID]: item.span.id } },
+                            { term: { [TRANSACTION_ID]: item.span.id } },
+                          ],
+                          minimum_should_match: 1,
+                        },
+                      },
+                    ],
+                  },
+                })),
+              },
+            },
+          ],
+        },
+      },
+    },
+  });
+  return response.hits.hits;
```

Comment:
I think this also needs a filter. If you're getting the details of a linked span that is a transaction, this search will return spans that have that transaction as a parent. I think... :) Let's write some tests for that scenario (and having parents of multiple spans in different traces, or children in different traces, etc).

Reply:
you're right, I used ProcessorEvent to fix it. The query looks like this now, which I think solves the problem.

```
filter: [
  { term: { [TRACE_ID]: item.trace.id } },
  { term: { [PROCESSOR_EVENT]: processorEvent } },
  ...(processorEvent === ProcessorEvent.transaction
    ? [{ term: { [TRANSACTION_ID]: item.span.id } }]
    : [{ term: { [SPAN_ID]: item.span.id } }]),
],
```

Reply:
So I'm filtering out spans that may be returned because of the transaction id. https://github.com/elastic/kibana/pull/126630/commits/651bf45f68ea0236a889afeb760d73586821374d#diff-d5848b1d2540e4a7d7f22749ef3ba6d1088820a67dc4e68958c6ef3940abd48eR109-R112